knowledge/pkg/vectorstore/pgvector/pgvector.go

```diff
@@ -349,7 +351,18 @@ func (v VectorStore) AddDocuments(ctx context.Context, docs []vs.Document, colle
 		return nil, sharedErr
 	}
 
-	return ids, v.conn.SendBatch(ctx, b).Close()
+	slog.Debug("Sending batch to pgvector", "store", "pgvector", "batchSize", b.Len())
+
+	results := v.conn.SendBatch(ctx, b)
+	defer results.Close()
```

Comment:
I don't think we want to defer this, since we call it on the same line as `return` later, and there are no other return statements prior to then.

Reply:
True, I removed that other return.. Close() is idempotent here, so it didn't do any harm :sweat_smile: